cmd/build/v2/services_test.go

```diff
@@ -175,24 +191,222 @@ func TestGetToBuildTag(t *testing.T) {
 			buildInfo: &model.BuildInfo{
 				Dockerfile: "Dockerfile",
 				Context:    ".",
 				Image:      "okteto.dev/test:test",
 			},
 			manifestName: "test",
 			svcName:      "test",
-			output:       "okteto.dev/test:test",
+			output:       []string{"okteto.dev/test:test"},
+		},
+		{
+			name: "access to global",
+			buildConfig: fakeConfig{
+				hasAccess: true,
+			},
+			buildInfo: &model.BuildInfo{
+				Dockerfile: "Dockerfile",
+				Context:    ".",
+			},
+			manifestName: "test",
+			svcName:      "test",
+			output:       []string{"okteto.dev/test-test:okteto"},
+		},
+		{
+			name: "access to global and isClean",
+			buildConfig: fakeConfig{
+				hasAccess: true,
+				isClean:   true,
+				sha:       "hello-this-is-a-test",
+			},
+			buildInfo: &model.BuildInfo{
+				Dockerfile: "Dockerfile",
+				Context:    ".",
+			},
+			manifestName: "test",
+			svcName:      "test",
+			output: []string{
+				"okteto.global/test-test:hello-this-is-a-test",
+				"okteto.dev/test-test:hello-this-is-a-test",
+				"okteto.global/test-test:okteto",
+				"okteto.dev/test-test:okteto",
+			},
+		},
+		{
+			name: "access to global and isClean with volumes",
+			buildConfig: fakeConfig{
+				hasAccess: true,
+				isClean:   true,
+				sha:       "hello-this-is-a-test",
+			},
+			buildInfo: &model.BuildInfo{
+				Dockerfile: "Dockerfile",
+				Context:    ".",
+				Image:      "nginx",
+				VolumesToInclude: []model.StackVolume{
+					{
+						LocalPath:  "",
+						RemotePath: "",
+					},
+				},
+			},
+			manifestName: "test",
+			svcName:      "test",
+			output: []string{
+				"okteto.global/test-test:hello-this-is-a-test",
+				"okteto.dev/test-test:hello-this-is-a-test",
+				"okteto.global/test-test:okteto-with-volume-mounts",
+				"okteto.dev/test-test:okteto-with-volume-mounts",
+			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			result := getToBuildTag(tt.manifestName, tt.svcName, tt.buildInfo)
+			ob := OktetoBuilder{
+				Config: tt.buildConfig,
+			}
+			result := ob.tagsToCheck(tt.manifestName, tt.svcName, tt.buildInfo)
 			require.Equal(t, tt.output, result)
 		})
 	}
 }
+
+func TestCheckIfCommitIsAlreadyBuilt(t *testing.T) {
+	type config struct {
+		cfg        OktetoBuilderConfigInterface
+		cmdOptions types.BuildOptions
+	}
+	type expected struct {
+		image     string
+		hasAccess bool
+	}
+	tests := []struct {
+		name     string
+		config   config
+		expected expected
+	}{
+		{
+			name: "no access",
+			config: config{
+				cfg: fakeConfig{
+					isClean:   true,
+					hasAccess: false,
+					sha:       "",
+				},
+			},
+			expected: expected{
+				image:     "",
+				hasAccess: false,
+			},
+		},
+		{
+			name: "no clean commit",
+			config: config{
+				cfg: fakeConfig{
+					isClean:   false,
+					hasAccess: true,
+					sha:       "",
+				},
+			},
+			expected: expected{
+				image:     "",
+				hasAccess: false,
+			},
+		},
+		{
+			name: "no access no clean commit",
```

Comment:
duplicated test with the above one? I think behavior is the same

Reply:
No, in this test the user doesn't have access to global registry while in the above the user has it.

I considered removing it because of the reason that you said but decided to have it in case we change the behavior still have the same output

Reply:
Even if the output is the same, the input is different. You don't know if `hasAccess` might change or not the output. From the point of view of this test, the implementation is a black box and it should check that given some inputs what are the expected output, so it should check different output combinations